ssg/build_remediations.py

```diff
@@ -346,8 +346,7 @@ def _get_rule_reference(self, ref_class):
     def update_when_from_rule(self, to_update):
         additional_when = ""
         if self.associated_rule.platform == "machine":
-            additional_when = ('ansible_virtualization_role != "guest" '
-                               'or ansible_virtualization_type != "docker"')
+            additional_when = ('ansible_virtualization_role not in container_list')
```

Comment:
When generating the Ansible Playbook with `oscap xccdf generate fix --fix-type ansible --profile pci-dss --output my.yml ./ssg-rhel8-ds.xml`, the `container_list` var is not defined.
For this use case to work, every Ansible snippet that add the `when` clause for `container_list`, also needs to define it.

There exists a syntax to make Ansible variables out of XCCDF variables, `xccdf-var`, consumed [here](https://github.com/ComplianceAsCode/content/blob/6a25c66822e34abc6aaa7346c362cb48032784ca/ssg/build_remediations.py#L559).
It takes an XCCDF Var and puts it into a syntax that openscap understands, so that it can handle it correctly when generating an Ansible fix.
Maybe there could be a variant of this that would just place a fixed value, `container-list-var`?

With the above, injecting `container_list` in `build_profile_remediations.py` with `add_extra_vars()`, may not be needed, as it would be handled like the other Ansible variables (testing needed).

Reply:
Interesting - so If I'm understanding you correctly, there is a potential to make this (or future additional vars) more consumable by turning it into an XCCDF var? I'll look more into this today. Thanks for pointing this out.

Reply:
Not necessarily an XCCDF Var, but it can be added statically.